tests/core/unittests/test_experiment.py

```diff
@@ -0,0 +1,423 @@
+        assert exp.algorithms == exp_config[0][1]['algorithms']
+        with pytest.raises(AttributeError):
+            exp.this_is_not_in_config = 5
+
+
+@pytest.mark.usefixtures("create_db_instance", "with_user_tsirif")
+class TestConfigProperty(object):
+    """Get and set experiment's configuration, finilize initialization process."""
+
+    def test_get_before_init_has_hit(self, exp_config, random_dt):
+        """Return a configuration dict according to an experiment object.
+
+        Assuming that experiment's (exp's name, user's name) has hit the database.
+        """
+        exp = Experiment('supernaedo2')
+        exp_config[0][1].pop('_id')
+        assert exp.configuration == exp_config[0][1]
+
+    def test_get_before_init_no_hit(self, exp_config, random_dt):
+        """Return a configuration dict according to an experiment object.
+
+        Before initialization is done, it can be the case that the pair (`name`,
+        user's name) has not hit the database. return a yaml compliant form
+        of current state, to be used with :mod:`metaopt.resolve_config`.
+        """
+        exp = Experiment('supernaekei')
+        cfg = exp.configuration
+        assert cfg['name'] == 'supernaekei'
+        assert cfg['refers'] is None
+        assert cfg['metadata']['user'] == 'tsirif'
+        assert cfg['metadata']['datetime'] == random_dt
+        assert len(cfg['metadata']) == 2
+        assert cfg['pool_size'] is None
+        assert cfg['max_trials'] is None
+        assert cfg['status'] is None
+        assert cfg['algorithms'] is None
+        assert len(cfg) == 7
+
+    def test_good_set_before_init_hit_with_diffs(self, exp_config):
+        """Trying to set, and differences were found from the config pulled from db.
+
+        In this case:
+        1. Force renaming of experiment, prompt user for new name.
+        2. Fork from experiment with previous name. New experiments refers to the
+           old one, if user wants to.
+        3. Overwrite elements with the ones from input.
+
+        .. warning:: Currently, not implemented.
+        """
+        new_config = {'metadata': {'user_version': 1.2}}
+        exp = Experiment('supernaedo2')
+        with pytest.raises(NotImplementedError):
+            exp.configure(new_config)
+
+    def test_good_set_before_init_hit_no_diffs_exc_max_trials(self, exp_config):
+        """Trying to set, and NO differences were found from the config pulled from db.
+
+        Everything is normal, nothing changes. Experiment is resumed,
+        perhaps with more trials to evaluate (an exception is 'max_trials').
+        """
+        exp = Experiment('supernaedo2')
+        # Deliver an external configuration to finalize init
+        exp_config[0][1]['max_trials'] = 5000
+        exp_config[0][1]['status'] = 'pending'
+        exp.configure(exp_config[0][1])
+        assert exp.configuration == exp_config[0][1]
+        exp_config[0][1]['max_trials'] = 1000  # don't ruin resource
```

Comment:
For this test and the following one, what are the last two lines for?

Reply:
`exp_config` is shared across tests and is a dictionary. So if I hack it to prove a certain point, I don't want to ruin whatever it was for the other tests.

Reply:
Can you make it such that each unit test copy it locally automatically? Otherwise it is very error prone. I would name the global one default_exp_config or global_exp_config such that it's clear that any modification is **not** applied on global_exp_config, but exp_config or local_exp_config.